reactioncommerce/api-plugin-email
# refactor: add new createdAt and updatedAt fields for Emails collection

Signed-off-by: Mohan Narayana <mohan.narayana@mailchimp.com>

Resolves #3 
Impact: **major**
Type: *refactor**

## Issue
The current Email plugin does not store the time and date when a document in the Emails Collection has been created or updated.

## Solution
The fields are added to the current schema. There is also a migration track created to add the new fields to the existing Emails collections.


## Testing
`{
  emailJobs(shopIds:["<shopId>"]){
    totalCount
    nodes{
      _id
      createdAt
      updatedAt
      updated
      data{
        subject
      }
    }
  }
}`
1. Create an order to trigger an email. Run the above query (before pulling the fix) to see the "Cannot query field createdAt" message on GraphQL playground.
2. Run [this](https://github.com/reactioncommerce/api-migrations/pull/35) migration to add the new fields to the DB. This way you will not see a "Value null" message when you run the query next on GQL.
3. Pull this fix and create another order. Ensure to sym-link the plugin. Running the above query again should fetch you all the data without errors.
4. Any existing records (before this schema change) will have 01-01-1970 as the createdAt and updatedAt values.
